firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteSchema.java

```diff
@@ -530,6 +537,26 @@ private boolean tableContainsColumn(String table, String column) {
     return columns;
   }
 
+  private void rewriteCanonicalIds() {
+    new SQLitePersistence.Query(db, "SELECT target_id, target_proto FROM targets")
```

Comment:
Dumb question: this file doesn't appear to employ a transaction anywhere. Doesn't `db.execSQL` auto-commit if you don't? It seems like it would be bad if this partially committed.

Reply:
The transaction is opened here: https://cs.corp.google.com/aosp-master/frameworks/base/core/java/android/database/sqlite/SQLiteOpenHelper.java?q=f:SQLiteopenhelper.java&dr&l=409

It will only get committed if line 421 executes.